lib/logstash/outputs/kafka.rb

```diff
@@ -250,27 +250,35 @@ def retrying_send(batch)
         begin
           # send() can throw an exception even before the future is created.
           @producer.send(record)
-        rescue org.apache.kafka.common.errors.TimeoutException => e
+        rescue org.apache.kafka.common.errors.RetriableException => e
+          logger.info("KafkaProducer#send() failed, will retry", :exception => e)
           failures << record
           nil
-        rescue org.apache.kafka.common.errors.InterruptException => e
```

Comment:
:red_circle: change in behavior we need to understand: `InterruptException` were previously retried now they aren't

Reply:
Good point! [`InterruptException`](https://kafka.apache.org/24/javadoc/org/apache/kafka/common/errors/InterruptException.html) gets thrown "if the thread is interrupted while blocked" according to the [KafkaProducer#send() Javadoc](https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#send-org.apache.kafka.clients.producer.ProducerRecord-org.apache.kafka.clients.producer.Callback-). I don't think this should happen unless the producer is stopped or otherwise interrupted.

Reply:
The current behaviour was introduced in https://github.com/logstash-plugins/logstash-output-kafka/pull/151, which doesn't motivate why `InterruptException` is in there.

Reply:
Thanks Mark, from that conversation it seemed like just following producer's [`send` throws](https://kafka.apache.org/0100/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#send(org.apache.kafka.clients.producer.ProducerRecord,%20org.apache.kafka.clients.producer.Callback))
... now Kafka's `InterruptException` doesn't inherit from `RetriableException` but I think we should keep the backwards compatible behavior esp. unless we clearly understand most cases for potential interrupts are not recoverable.

Reply:
I've explicitly added `InterruptException` back in the retry logic. I'm still not sure this is the best way to go (maybe rethrowing it as an `InterruptedException` would make more sense), but at least the behaviour is backwards compatible now.